pallets/identity/src/lib.rs

```diff
@@ -1773,10 +1786,23 @@ impl<T: Trait> Module<T> {
         issuer: IdentityId,
         scope: Option<Scope>,
     ) -> DispatchResult {
+        // Ensure the target is the issuer of the claim.
         ensure!(
             target == issuer,
             Error::<T>::ConfidentialScopeClaimNotAllowed
         );
+        let (pk, sk) = Self::get_claim_keys(target, claim_type, issuer, scope.clone());
+        if let IdentityClaim {
+            claim: Claim::InvestorUniqueness(_, scope_id, _),
+            ..
+        } = Claims::get(&pk, &sk)
+        {
+            // Ensure that the target has balance at scope = 0.
+            ensure!(
+                T::AssetSubTraitTarget::balance_of_at_scope(&scope_id, &target) == Zero::zero(),
+                Error::<T>::TargetHasNonZeroBalanceAtScopeId
+            );
+        }
         Self::base_revoke_claim(target, claim_type, issuer, scope);
         Ok(())
     }
```

Comment:
I'd suggest merging `revoke_confidential_scope_claim` and `base_revoke_claim`. There seems to be no advantage of splitting them.